File: Veterinario/VeterinarioBlueJ/ChequeoMedico.java
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class ChequeoMedico {
    private String idConsulta;
    private Date fecha;
    private Mascota paciente;
    private Veterinario especialista;
    private String diagnostico;
    private List<DetalleReceta> receta;

    public ChequeoMedico(String idConsulta, Mascota paciente, Veterinario especialista, String diagnostico) {
        this.idConsulta = idConsulta;
        this.fecha = new Date();
        this.paciente = paciente;
        this.especialista = especialista;
        this.diagnostico = diagnostico;
        this.receta = new ArrayList<>();
    }

    public void agregarMedicamento(DetalleReceta detalle) {
        this.receta.add(detalle);
    }

    public String getIdConsulta() { return idConsulta; }
    public Date getFecha() { return fecha; }
    public Mascota getPaciente() { return paciente; }
    public Veterinario getEspecialista() { return especialista; }
    public String getDiagnostico() { return diagnostico; }
    public List<DetalleReceta> getReceta() { return receta; }
}

File: Veterinario/ConsultorioVeterinario-V2/ConsultorioVeterinario/Logica/ChequeoMedico.cs
using System;
using System.Collections.Generic;

namespace ConsultorioVeterinario.Logica
{
    public class ChequeoMedico
    {
        public string IdConsulta { get; set; }
        public DateTime Fecha { get; set; }
        public Mascota Paciente { get; set; }
        public Veterinario Especialista { get; set; }
        public string Diagnostico { get; set; }
        public List<DetalleReceta> Receta { get; set; } = new List<DetalleReceta>();

        public ChequeoMedico(string id, Mascota paciente, Veterinario especialista, string diagnostico)
        {
            IdConsulta = id;
            Fecha = DateTime.Now;
            Paciente = paciente;
            Especialista = especialista;
            Diagnostico = diagnostico;
        }
    }
}
